### PR TITLE
intel: drivers: dmic refactor - move DMIC_HW_IOCLK to Kconfig

### PR DESCRIPTION
**Pull request by aborisovich**
Moved hardcoded cycles values for DMIC IO from dmic.h header
to per-platform configuration file (defconfig files for
non-Zephyr builds and app/boards for Zephyr builds).
Added new entry to intel drivers Kconfig - CONFIG_DMIC_HW_IOCLK.
Modification of this clock value may be used for testing purposes
like building firmware for FPGA or simulator.

Signed-off-by: Andrey Borisovich <andrey.borisovich@intel.com>

### CODE REVIEW

**Review by tmleman — APPROVED** [review]

File: src/drivers/intel/dmic/dmic_computed.c [hunk 1]
```diff
@@ -67,7 +67,7 @@ static void find_modes(struct dai *dai,
 
 	/* Check for sane pdm clock, min 100 kHz, max ioclk/2 */
 	if (dmic->global->prm[di].pdmclk_max < DMIC_HW_PDM_CLK_MIN ||
-	    dmic->global->prm[di].pdmclk_max > DMIC_HW_IOCLK / 2) {
+	    dmic->global->prm[di].pdmclk_max > CONFIG_DMIC_HW_IOCLK / 2) {
```
Comment (lgirdwood) [inline src/drivers/intel/dmic/dmic_computed.c:70]: Is there a PR to do this in Zephyr DMIC too ?

**Review by lgirdwood — APPROVED** [review]

File: src/drivers/intel/dmic/dmic_computed.c (hunk 1 above)
Comment (aborisovich) [reply src/drivers/intel/dmic/dmic_computed.c:70]: Zephyr DMIC already has configurable HW_IO_CLOCK (CONFIG_DAI_DMIC_HW_IOCLK).

**Comment by lgirdwood** [comment]
@aborisovich can you check CI, looks like this PR is failing some DMIC tests, could be the Kconfig default is wrong for JSL.

**Comment by aborisovich** [comment]
> @aborisovich can you check CI, looks like this PR is failing some DMIC tests, could be the Kconfig default is wrong for JSL.

Yes, I've skipped it intentionally but did not see that it actually uses CONFIG_ICELAKE=y. Fixed.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by aborisovich** [comment]
Can we merge? Are there any more comments from reviewers?

**Pull request merged**